app/models/setting.rb

```diff
@@ -242,9 +243,16 @@ def self.create_existing(s, opts)
       attrs = column_check([:default, :description, :full_name, :encrypted])
       to_update = Hash[opts.select { |k, v| attrs.include? k }]
       to_update[:value] = readonly_value(s.name.to_sym) if s.has_readonly_value?
-      s.update(to_update)
+      # default is converted to yaml so we need to convert the yaml here too,
+      # in order to check, if the update of default is needed
+      # if it is the same, we don't try to update default, it would trigger
+      # update query for every setting
+      to_update.delete(:default) if to_update[:default].to_yaml.strip == s[:default]
+      s.attributes = to_update
+      s.save if s.changed?
       s.update_column :category, opts[:category] if s.category != opts[:category]
-      s.update_column :full_name, opts[:full_name] unless column_check([:full_name]).empty?
+      # we only update category (skipping validations) if it really differs
+      s.update_column :full_name, opts[:full_name] if s.full_name != opts[:full_name] && column_check([:full_name]).present?
```

Comment:
This always triggered update no matter if the full_name has changed. While this is normally updated by the call above, I suppose we did it here using `update_column` explicitly because we wanted to skip validation. We now check if this is really needed to avoid unnecessary update query for each setting